packages/paypal-express-integration/src/paypal-express-button-strategy.ts

```diff
@@ -0,0 +1,181 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { pick } from 'lodash';
+
+import {
+    CheckoutButtonInitializeOptions,
+    CheckoutButtonStrategy,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+    PaymentIntegrationService,
+    StandardError,
+    INTERNAL_USE_ONLY,
+    SDK_VERSION_HEADERS,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+import {
+    PaypalActions,
+    PaypalAuthorizeData,
+    PaypalButtonStyleShapeOption,
+    PaypalButtonStyleSizeOption,
+    PaypalClientToken,
+    PaypalSDK,
+} from './paypal-express-types';
+import {
+    PaypalExpressButtonInitializeOptions,
+    WithPaypalExpressButtonInitializeOptions
+} from './paypal-express-button-initialize-options';
+import PaypalScriptLoader from './paypal-express-script-loader';
+
+export default class PaypalButtonStrategy implements CheckoutButtonStrategy {
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private paypalExpressScriptLoader: PaypalScriptLoader,
+        private formPoster: FormPoster,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithPaypalExpressButtonInitializeOptions,
+    ): Promise<void> {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paypalSDK = await this.paypalExpressScriptLoader.loadPaypalSDK(
+            paymentMethod.config.merchantId,
+        );
+
+        this.renderButton(options, paypalSDK);
+    }
+
+    private renderButton(
+        options: CheckoutButtonInitializeOptions & WithPaypalExpressButtonInitializeOptions,
+        paypalSDK: PaypalSDK,
+    ) {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
```

Comment:
Do we still need this condition? You are checking the same in initialize method

Reply:
removed